tests/ecosystem/upgrade/test_upgrade.py

```diff
@@ -129,9 +132,36 @@ def test_upgrade():
         resource_name=constants.OPERATOR_CATALOG_SOURCE_NAME,
         namespace=constants.MARKETPLACE_NAMESPACE,
     )
+    version_before_upgrade = config.ENV_DATA.get("ocs_version")
+    upgrade_version = config.UPGRADE.get(
+        "upgrade_ocs_version", version_before_upgrade
+    )
+    parsed_version_before_upgrade = parse_version(version_before_upgrade)
+    parsed_upgrade_version = parse_version(upgrade_version)
+    assert parsed_upgrade_version >= parsed_version_before_upgrade, (
```

Comment:
I answered this to Coady's question. In case of 4.2.1 upgrade to 4.2.2 we still are using same OCS version config file for 4.2 and hence OCS version is just X.Y there so we are still on 4.2 and upgrading just Z stream, so no need to fail for that case.